### ISSUE TITLE
Remove legacy py-struct provider compatiblity code

### ISSUE THREAD

**Issue by rickeylev** [bazelbuild/bazel#15353]
### Description of the feature request:

Remove the compatibility code for handling the legacy py-struct provider. This was gated by the `--incompatible_disallow_legacy_py_provider` flag, and has been waiting for Google to enable that. It's been enabled by default in Bazel since ~2019-ish.

Google's finally enabled that flag, so the code dealing with the legacy py-struct provider can be removed.

This is primarily PyProviderUtils, I think.

### What underlying problem are you trying to solve with this feature?

Code cleanup

### Which operating system are you running Bazel on?

n/a

### What is the output of `bazel info release`?

n/a

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

n/a

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?

```text
n/a
```

### Have you found anything relevant by searching the web?

n/a

**Comment by sgowroji** [comment]
Hello @rickeylev, Found there is already a tracking bug with the same above details https://github.com/bazelbuild/bazel/issues/7741. Please correct me if am wrong if the above Feature request is different to it. Thanks!

**Comment by rickeylev** [comment]
Yes, that's basically the same. Not sure how I missed that, sorry.
